pkg/reconciler/internal/hook/hook_test.go

```diff
@@ -252,6 +253,23 @@ var _ = Describe("Hook", func() {
 	})
 })
 
+// validateSourceHandlerType takes in a source.Source and uses reflection to determine
+// if the handler used by the source matches the expected type.
+// It is assumed that the source.Source was created via the source.Kind() function.
+func validateSourceHandlerType(s source.Source, expected interface{}) bool {
+	sourceVal := reflect.Indirect(reflect.ValueOf(s))
+	handlerFieldVal := sourceVal.FieldByName("Handler")
```

Comment:
Is it possible for this to return `nil`?

If so, we should do `nil` checks, and perhaps change the function signature to return `error` instead of `bool` such that we can return some meaningful information when the validation fails.

Reply:
I don't believe it will return the literal value `nil`, but there is a [`value.IsNil()`](https://pkg.go.dev/reflect#Value.IsValid) method on the returned type.

I do think returning an error instead of a boolean is a good idea. Will update now

Reply:
Updated to return an error instead of boolean

Reply:
What if there is no `Handler` field? Seems like we should have an explicit error for that.

Reply:
I've updated this to use the `value.IsValid()` function to validate that the return value is value and we can perform validations with it. Also added some simple tests to ensure that it is functioning as expected